src/pkg/cli/compose/validation.go

```diff
@@ -29,6 +39,11 @@ func ValidateProject(project *composeTypes.Project) error {
 	sort.Slice(services, func(i, j int) bool {
 		return services[i].Name < services[j].Name
 	})
+
+	if err := ValidateProjectConfig(context.Background(), project, listConfigNamesFunc); err != nil {
```

Comment:
`context.TODO()` because we really should be using the caller's context here. `context.Background()` is fine in some corner cases, and in test code.